lightning/src/ln/channelmanager.rs

```diff
@@ -2098,6 +2101,11 @@ where
 	///
 	/// May generate a [`SendShutdown`] message event on success, which should be relayed.
 	///
+	/// Raises [`APIError::ChannelUnavailable`] if the channel cannot be closed due to failing to
+	/// generate a shutdown scriptpubkey or destination script set by
+	/// [`SignerProvider::get_shutdown_scriptpubkey`] and [`SignerProvider::get_destination_script`].
```

Comment:
IIUC I don't think `SignerProvider::destination_script` is used when closing?

Reply:
good catch, fixed some of the docs